web/turkey/docs/build/html/_sources/extending.txt

```diff
@@ -0,0 +1,156 @@
+For each scrolling event, reports the end position, the raw change in pixels from the previous position, and the timestamp of the event. Records data for both horizontal and vertical scrolling, debounced.
+
+TIME BASED AUDITORS
++++++++++++++++++++
+Auditors recording the time spent completing the task.
+
+Total Task Time
++++++++++++++++
+The total time it takes the worker to complete this task in milliseconds.
+
+On Focus Time
++++++++++++++
+The total amount of time in milliseconds the worker spent on focus (not tabbed out, etc). Also a `Focus Based Auditor <extending.html#focus-based-auditors>`_.
+
+Recorded Time Disparity
++++++++++++++++++++++++
+The total amount of time in milliseconds the worker spent off focus (tabbed out, etc). Also a `Focus Based Auditor <extending.html#focus-based-auditors>`_.
+
+OTHER AUDITORS
+++++++++++++++
+Pasting events, User Agent spying, and URL.
+
+Pastes Total
+++++++++++++
+Total number of pastes actions.
+
+Pastes Specific
++++++++++++++++
+The content (text) of a paste action.
+
+URL
++++
+Returns the URL of the current page.
+
+User Agent
+++++++++++
+Returns the User Agent of the worker; can be parsed to find browser/version and operating system/version.
+
+Writing Your Own
+----------------
+Javascript: The Actual Auditor
+++++++++++++++++++++++++++++++
+Auditors generally have the same format, and referencing the default auditors can be helpful when trying to write your own. Create a JavaScript auditor object (.../js/auditors/) with your needed fields, a function that modifies those fields, and a function to submit those fields. The modifier function is the action that should be executed when your auditor detects a certain interaction. The submission function ("submit_callable") must return a JavaScript object or array of JavaScript objects. The fields can have the value of a single value (int, string, etc). Be sure to register your auditor to the "overlord" object (.../js/submission_handler.js), which handles the submission of results and all auditor data when the submit button is clicked.
+
+Python: The Auditor's Models
+++++++++++++++++++++++++++++
+Similarly to the JavaScript side, the Python code (.../survey/auditors.py) across all auditors is rather similar and should be helpful when writing the models for your auditor. Firstly, add your auditor to the "NAME_TO_AUDITOR" dictionary at the top. You will then need to write a class for your auditor's data and for the auditor itself. The data class will have a "general_models" field with the foreign key being the name of your auditor and a field corresponding to each item specified in the auditor's returned object. For instance, if your auditor returned { 'count': some_int_var }, this class must have a "count" field that accepts data of type int. The actual auditor class has a "script_location" field, which is the directory and name of your auditor file, and a "data_model" field, which is simply the data class you just wrote.
```

Comment:
Again, getting into backend processing and overriding somewhere is probably necessary